modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParser.java

```diff
@@ -136,4 +145,38 @@ public LicenseModelProperty getLicenseModelProperty() {
 
     return additionalModelProperties;
   }
+
+  @Override
+  public Optional<DeprecationModel> getDeprecationModel() {
+    return JavaExtensionModelParserUtils.getDeprecationModel(extensionElement);
+  }
+
+  @Override
+  public XmlDslModel getXmlDslModel() {
+    String extensionName = extensionElement.getName();
```

Comment:
Same comment as in the interface definition, maybe it should just provide the annotations information.